syntax = "proto3";

service CatalogService {
    rpc Lookup (LookupRequest) returns (LookupResponse);
    rpc Trade (TradeRequest) returns (TradeResponse);
}

message LookupRequest {
    string stock_name = 1;
}
message LookupResponse {
    string name = 1;
    float price = 2;
    int32 quantity = 3;
    int32 code = 4;
    string message = 5;
}
message TradeRequest {
    string name = 1;
    int32 number_of_items = 2;
    string type = 3;  
}
message TradeResponse {
    int32 code = 1; 
}
